context.go

```diff
@@ -297,7 +297,31 @@ func (c *Context) GetInt(key string) (i int) {
 	return
 }
 
-// GetInt64 returns the value associated with the key as an integer.
+// GetInt8 returns the value associated with the key as an integer 8.
+func (c *Context) GetInt8(key string) (i8 int8) {
```

Comment:
Maybe we should keep the same logic as the original `.Get` and returns a `(int8, bool)` to avoid panicking in case the value if not of the asked type. This comment applies to other methods as well. What do you think?

Reply:
Yes, you are right, I just noticed this problem too, but there is a small compatibility problem that seems strange, the methods I added this time are styled by the functions that already existed before: 
```go
func (c *Context) GetInt(key string) (i int)
```
and
```go
func (c *Context) GetInt64(key string) (i64 int64)
```
The style is kept uniform, and these two functions seem to start from this PR:
```text
https://github.com/gin-gonic/gin/pull/2425
```
Then other GetXxx functions such as:
```go
func (c *Context) GetString(key string) (s string)
func (c *Context) GetBool(key string) (b bool)
```
It seems that they all follow the rule of only one return value. 

All in all, there seem to be two styles for the Get method on the Context. The first one is:
```go
func (c *Context) Get(key string) (value any, exists bool)
```
Represented by two return values, the other is based on
```go
func (c *Context) GetInt(key string) (i int)
```
Etc is represented by a return value

![image](https://github.com/gin-gonic/gin/assets/12819457/721622a7-163c-4437-aa06-fd46e3057f0b)

Reply:
You're right, I didn't see that, so what you've done makes sense.

But for the record, I guess we could leverage Go generic with something like this:

```go
func Get[T any](c *gin.Context, key string) (r T) {
	if val, ok := c.Get(key); ok && val != nil {
		r, _ = val.(T)
	}
	return
}
```

Sure we can't add this on the `Context` type because Go doesn't currently support method level generic types yet.

Reply:
Yes, you have a point, the best way is to add a generic method to Context, although I don't think Golang will support this feature any time soon, but I believe it will be supported in the future, it is a mature language should have a feature. some proposal about it:
[https://github.com/golang/go/issues/49085](https://github.com/golang/go/issues/49085)

However, in the absence of generics, enumeration method can only be used. Although there is a lot of repetitive work, using a separate method can solve this problem, but it will be a problem to promote the use of this function. Most people use the method in Context and then view it with dot. So I think it is necessary to add these methods to Context for now, as a transition that will remove in the future when Golang supports Struct generic methods.